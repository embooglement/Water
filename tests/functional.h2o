let test_bind = func() {
	let f = func(x, y, z) {
		println(x, y, z);
	};

	let g = bind(f, 1, 2);
	g(3);
};

let test_constant = func() {
	let f = constant("hello world");
	println(f(1, 2, 3));
};

let test_compose = func() {
	let f1 = func(x) { 
		return x + 1;
	};

	let f2 = func(x) {
		return x - 1;
	};

	let f3 = func(x) {
		return x * 2;
	};

	let g = compose(f1, f2, f3);
	println(g(3));
};

let test_id = func() {
	let x = 2;
	println(id(x));
};

test_bind();
test_constant();
test_compose();
test_id();
