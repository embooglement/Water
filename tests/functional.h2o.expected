1 2 3
hello world
6
2

